stac_fastapi/pgstac/models/links.py

```diff
@@ -51,7 +51,11 @@ def base_url(self):
     @property
     def url(self):
         """Get the current request url."""
-        return str(self.request.url)
+        url = urljoin(str(self.request.base_url), self.request.url.path.lstrip("/"))
+        if qs := self.request.url.query:
+            url += f"?{qs}"
```

Comment:
This feels over engineered but it's the only way I could find to make it works 🤷